test/raft/replication.hh

```diff
@@ -407,7 +407,11 @@ public:
     future<> apply(const std::vector<raft::command_cref> commands) override {
         auto n = _apply(_id, commands, hasher);
         _seen += n;
-        if (n && _seen == _apply_entries) {
+        if (n && _seen >= _apply_entries) {
```

Comment:
Won't this problem cause premature _done? In case the retry bumps _seen to be equal to _apply_entries but there is still a command left to apply?

Reply:
No, the last commit fixes the problems that we know about.

There is a general theoretical problem with the replication_test framework that it's retrying commands on `commit_status_unknown`.

*If* there is a test scenario where `commit_status_unknown` may get thrown on a command that was actually committed, then we will see a build failure. But after this PR there will be no such known test scenarios. The PR fixes the backpressure test failures which we observed, and we didn't see any other replication_test failures.

Reply:
Ok. But this is still disturbing, theoretical problems tend to convert into real problems in the future.

Reply:
Yep, that's why I added the warnings to make it easier to debug any such problems if they appear in the future.
Not sure if worth opening an issue though. @kostja would probably disapprove because it would be an issue without evidence of actual failure/bug happening :P